lib/std/uniquelabels/unique_labels_test.go

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2025 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package uniquelabels
+
+import (
+	"bytes"
+	"encoding/json"
+	"testing"
+)
+
+func TestInternedLabelsJSONRoundTrip(t *testing.T) {
```

Comment:
I think it's worth filling out this test suite a bit more:

- Additional Equality testcases
- JSON marshal / unmarshal edge cases - e.g., for nil and Empty
- Test cases for `IntersectAndFilter`

Reply:
Yep, haven't fleshed out the tests yet.